### PR TITLE
Fix --mode for scalability jobs, and fix unit-test coverage

### PR DESCRIPTION
**Pull request by krzyzacy**
The test was missing Jenkins coverage when I flipped default mode from `docker` to `local`

/assign @shyamjvs

### CODE REVIEW

File: jenkins/bootstrap_test.py [hunk 1]
```diff
@@ -2295,6 +2291,12 @@ def test_valid_job_envs(self):
                                 )
                 if config[job]['scenario'] == 'kubernetes_e2e':
                     args = config[job]['args']
+                    if job in self.prowjobs:
+                        for arg in args:
+                            # --mode=local is default now
+                            self.assertNotIn('--mode', args, job)
+                    else:
+                        self.assertIn('--mode=docker', args, job)
```
Comment (shyamjvs) [inline jenkins/bootstrap_test.py]: Are we sure there are no local non-prow jobs?

Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: yes since the test passes :-)

File: jobs/config.json [hunk 2]
```diff
@@ -2368,7 +2368,8 @@
       "--test_args=--ginkgo.skip=\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\] --allowed-not-ready-nodes=20 --node-schedulable-timeout=30m --cluster-ip-range=10.160.0.0/13 --minStartupPods=8",
       "--gcp-project=kubernetes-scale",
       "--provider=gce",
-      "--gcp-zone=us-east1-a"
+      "--gcp-zone=us-east1-a",
+      "--mode=docker"
```
Comment (shyamjvs) [inline jobs/config.json]: It's already defined above.

Comment (krzyzacy) [inline jobs/config.json]: removed

**Comment by shyamjvs** [comment]
/lgtm
Thanks for fixing this.

**Comment by krzyzacy** [comment]
@spxtr I think gofmt and govet should only run if I have go changes?

**Pull request merged**